### PR TITLE
Use Rails 5-style detailed errors

### LINKED ISSUE #250: Replace symbolic errors with new Rails 5 type errors (i.e. details)
https://cowbell-labs.com/2015-01-22-active-model-errors-details.html

Should we go ahead and change our API to match what's coming?

### PR DESCRIPTION
**Pull request by tfausak**
Fixes #250.

### CODE REVIEW

**Comment by tfausak** [comment]
[This build](https://travis-ci.org/orgsync/active_interaction/jobs/59942806) failed because active_model-errors_details depends on activemodel >= 4.0, whereas active_interaction depends on activemodel ~> 3.2.

**Comment by tfausak** [comment]
I am now waiting on cowbell/active_model-errors_details#2 to add support for Rails 3.2.

**Comment by tfausak** [comment]
My pull request was merged and released as v1.2.0. However it requires ActiveModel >=3.2.13, whereas we require ActiveModel >=3.2.0.

**Comment by tfausak** [comment]
39f2dd7d314ee7bf43eb9179b8fb47703d4069af added 3.2.0-compatible support for the detailed errors.

**Comment by tfausak** [comment]
I realized that this is technically a backport, so I moved it into the backports file.

File: lib/active_interaction/backports.rb [hunk 1]
```diff
@@ -17,5 +17,53 @@ def []=(name, value)
...
+
+      CALLBACKS_OPTIONS = ::ActiveModel::Errors::CALLBACKS_OPTIONS
+      private_constant :CALLBACKS_OPTIONS
+
+      included do
+        attr_reader :details
+
+        %w[initialize initialize_dup add clear delete].each do |method|
+          alias_method "#{method}_without_details", method
+          alias_method method, "#{method}_with_details"
```
Comment (AaronLasseigne) [inline lib/active_interaction/backports.rb]: `alias_method_chain`?

Comment (tfausak) [inline lib/active_interaction/backports.rb]: :+1: 

At some point along the way, it was not working for me. Now it is. I am not sure what changed.

**Comment by AaronLasseigne** [comment]
We need to update the README to reflect these changes. (probably for some of the other 2.0 stuff as well)

**Comment by AaronLasseigne** [comment]
Code wise it's a :shipit:.

**Comment by tfausak** [comment]
I'll update the v2.0.0 documentation in a separate branch.

**Pull request merged**